app/extensions/__init__.py

```diff
@@ -79,6 +80,50 @@
 ##########################################################################################
 
 
+class ExtraValidationSchema(Schema):
```

Comment:
I'm generally liking this concept. It seems really flexible but I will admit that I don't follow what this part is doing.
Can you add some more comments to explain which bits are called automatically from within marshmallow and what it is that causes the 'errors' return from the schema.load call (which I'm guessing is what calls this) to be set and what it would be set to.

Reply:
Yeah I'll need to add more comments to explain what this is